doctr/models/_utils.py

```diff
@@ -154,3 +155,39 @@ def get_bitmap_angle(bitmap: np.array, n_ct: int = 20, std_max: float = 3.) -> f
             angle = 90 + angle
 
     return angle
+
+
+def rotate_boxes(
+    boxes: np.array,
+    angle: float = 0.,
+    min_angle: float = 1.
+) -> np.array:
+    """Rotate a batch of straight bounding boxes (xmin, ymin, xmax, ymax) of an angle,
+    if angle > min_angle, around the center of the page.
+
+    Args:
+        boxes: (N, 4) array of (relative) boxes
+        angle: angle between -90 and +90 degrees
+        min_angle: minimum angle to rotate boxes
+
+    Returns:
+        A batch of rotated boxes (N, 5): (x, y, w, h, alpha) or a batch of straight bounding boxes
+    """
+    # If small angle, return boxes (no rotation)
+    if abs(angle) < min_angle or abs(angle) > 90 - min_angle:
+        return boxes
+    # Compute rotation matrix
+    angle = angle * np.pi / 180.  # compute radian angle for np functions
+    R = np.array([[np.cos(angle), np.sin(angle)], [-np.sin(angle), np.cos(angle)]])
+    # Translate to relative origin of the page
+    boxes[:, [0, 2]] -= .5
+    boxes[:, [1, 3]] -= .5
```

Comment:
We have to be careful with this inplace modification: if the original straight boxes are used later, we will no longer have access to their original values